src/Orleans.Runtime/Silo/SiloLifecyclePerfReporter.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace Orleans.Runtime
+{
+    internal class SiloLifecyclePerfReporter
```

Comment:
Don't we want to have a public interface for this? I think it could be beneficial for others as well, our implementation could stay internal.

The public interface should contain the ```AddReport``` method and a property or method to get the added reports, but NOT the ```FlushReports```, which is our implementation.

FlushReports could be an extension method or something similar, but the ```SiloLifecyclePerfReporter``` class should be "as dump as possible" IMHO.

Reply:
It is unclear to me on the requirements of this feature, so in my mind, I wanted this feature as an internal feature first, and then we can work out the patterns in this feature by testing it in our internal silo lifecycle participants. If we feel the feature is mature, we can then release it as public.